### PR TITLE
Keep glib::clone spans

### LINKED ISSUE #287: clone! macro makes error messages cryptic
Hello,

when using the `clone!` macro rustc doesn't report the line where the errror occured correctly and instead just points to the entire macro invocation. This forces users of the macro to guess where the error occured, which isn't trivial when the function `clone!` encapsulates is longer than a few lines.

E.g. when I forget to put a semicolon somewhere the error message looks like this:

```
error: expected one of `.`, `;`, `?`, or an operator, found `self_`
   --> src/widgets/sync_list_box.rs:133:40
    |
133 |                   .connect_row_activated(clone!(@weak obj => move |list_box, row| {
    |  ________________________________________^
134 | |                     let self_ = imp::HealthSyncListBox::from_instance(&obj);
135 | |                     if (row == &self_.google_fit_start_sync_row.get()) {
136 | |                         self_.google_fit_stack.set_visible(true);
...   |
180 | |                     }
181 | |                 }));
    | |__________________^ expected one of `.`, `;`, `?`, or an operator
    |
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to previous error
```

So I have to check some 50 lines to know where I forgot the semicolon.

See https://github.com/Cogitri/clone-reproducer for a reproducer:

```
error[E0599]: no method named `qiut` found for struct `gtk4::Application` in the current scope
   --> src/bin/entry_completion.rs:101:9
    |
101 | /         glib::clone!(@weak application => move |_action, _parameter| {
102 | |             application.qiut();
103 | |         }),
    | |__________^ method not found in `gtk4::Application`
    |
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

In this case the error message seems OK because it's only a one line function, but ideally it'd point to only line 102 and not 101/103.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #287

### CODE REVIEW

File: glib-macros/src/clone.rs [hunk 1]
```diff
@@ -539,22 +539,66 @@ pub(crate) fn clone_inner(item: TokenStream) -> TokenStream {
     let return_kind = parse_return_kind(&mut parts);
     check_before_closure(&mut parts);
     let closure = get_closure(&mut parts);
-    let body = tokens_to_string(parts);
-
-    let x = format!(
-        "{{\n{}\nmove |{}| {{\n{}\nlet ____ret = {{ {} }};\n____ret\n}}\n}}",
```
Comment (sdroege) [inline glib-macros/src/clone.rs:545]: How did we ever think that was a good idea?

Comment (GuillaumeGomez) [reply glib-macros/src/clone.rs:545]: Well, it makes the reading easier?

**Comment by GuillaumeGomez** [comment]
@pbor: Any idea what's wrong with Windows CI here?

**Comment by sdroege** [comment]
```
 error: returning the result of a `let` binding from a block
   --> glib/tests/clone.rs:478:9
    |
477 |         let y = 19;
    |         ----------- unnecessary `let` binding
478 |         y
    |         ^
```

@GuillaumeGomez

**Comment by GuillaumeGomez** [comment]
Ah right, forgot to remove this one.

File: glib/tests/clone.rs [hunk 2]
```diff
@@ -400,11 +400,15 @@ fn test_clone_macro_double_rename() {
     closure(0i8); // to prevent compiler error for unknown `x` type.
     let _ = clone!(@weak v, @weak w as _x => @default-return true, move || false);
 
-    let closure = clone!(@strong v as _x, @strong w => @default-return true, move |_| { false });
+    let closure = clone!(@strong v as _x, @strong w => @default-return true, move |_| {
+        false
```
Comment (sdroege) [inline glib/tests/clone.rs:404]: Ah does rustfmt understand the body of the `clone!` macro now because we don't go through a string?

Comment (GuillaumeGomez) [reply glib/tests/clone.rs:404]: No idea, it was complaining about the unneeded parens and I wanted to them.

**Review by sdroege — APPROVED** [review]

**Pull request merged**